R/data-combined.R

```diff
@@ -0,0 +1,807 @@
+    #'     molWeight
+    #'     lloq
+    #'     Source
+    #'     Sheet
+    #'     Organ
+    #'     Compartment
+    #'     Molecule
+
+
+    toDataFrame = function() {
+      # It is deliberate that this "cleaning" happens every time the user calls
+      # this method. R6 has a reference semantics, i.e., the object will be
+      # modified in place and the combined data will be updated with it. But,
+      # for future updates, we do need to retain some information that is
+      # omitted during cleaning.
+      #
+      # For example, we remove offset and scale factor columns during cleaning,
+      # but they need to be retained in the private copy of the combined
+      # dataframe because the datasets that might be added in the future will
+      # have their own offsets and scale factors, which will need to be appended
+      # to the existing ones to be returned by the `dataTransformations()`
+      # active binding. If we were to clean the dataframe in place, this will
+      # not be possible since there won't be offset and scale factors columns to
+      # append to.
+      private$.cleanDF(private$.dataCombinedDF)
+    },
+
+    ## print method -----------------
+
+    #' @description
+    #' Print the object to the console
+
+    print = function() {
+      private$printClass()
+
+      # TODO: what do we want to print here?!
+
+      # for method chaining
+      invisible(self)
+    }
+  ),
+
+  # active bindings ---------------------------------------------------
+
+  active = list(
+
+    #' @field names A vector of unique names of `DataSet` objects and/or
+    #'   quantities or paths from `SimulationResuls` object sorted
+    #'   alphabetically.
+
+    # just a way to access whatever was specified
+    names = function(value) {
+      if (missing(value)) {
+        private$.names
+      } else {
+        stop(messages$errorPropertyReadOnly(
+          "names",
+          optionalMessage = "Names are assigned using `names` argument in `$addSimulationResults()` or `$addDataSets()` methods."
+        ))
+      }
+    },
+
+    #' @field groupMap A dataframe specifying which datasets have been grouped
+    #'   together and the name and the nature (observed or simulated?) of the
+    #'   data. If a dataset was not assigned to any group, this is denoted by
+    #'   `NA` in the dataframe.
+
+    # just a way to access whatever was specified
+    groupMap = function(value) {
+      if (missing(value)) {
+        private$.groupMap
+      } else {
+        stop(messages$errorPropertyReadOnly(
+          "groupMap",
+          optionalMessage = "Data sets are assigned to groups when adding via `$addSimulationResults()` or `$addDataSets()` methods."
+        ))
+      }
+    },
+
+    #' @field dataTransformations A dataframe specifying which offsets and scale
+    #'   factor values were specified by the user for each dataset.
+
+    # just a way to access whatever was specified
+    dataTransformations = function(value) {
```

Comment:
Should be `dataTransfors` to be consistent with the setter.

Reply:
I am actually going the other way: changing the method name to `setDataTransformations`. 

_Transforms_ sounds awkward. Better to be verbose here.